woocommerce/woocommerce
# Try: Use Zod validation to generate JSON Schema

### Changes proposed in this Pull Request:

* Adds Zod to the client and a couple of basic product fields for example
* Converts the Zod schema to a JSON Schema for use on the server

<img width="557" alt="Screenshot 2024-05-30 at 1 13 29 PM" src="https://github.com/woocommerce/woocommerce/assets/10561050/8f51c2c0-39e4-4b9b-9d71-8a268bbbe566">




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Navigate to the new product editing experience
2. Open your console
3. See the JSON Schema
4. See the validation results on typing in the name and price fields (sale price should be less than regular price)



### Changelog entry







<details>

#### Significance




#### Type




#### Message 


</details>

<details>

#### Comment 


</details>
